src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROW_COEFFICIENT = 1.5;
+    private static final String ADD = "ADD";
+    private static final String NO_ADD = "noADD";
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexValid(index, "ADD");
+        resize();
+        System.arraycopy(elementData, index,elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i), size);
```

Comment:
```suggestion
            add(list.get(i));
```

i think it's better to use `add(T value)` method here

Reply:
exactly)